topic-operator/src/test/java/io/strimzi/operator/topic/KafkaStreamsTopicStoreClusterTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.topic;
+
+import io.vertx.core.Promise;
+import io.vertx.junit5.Checkpoint;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.Collections;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@ExtendWith(VertxExtension.class)
+public class KafkaStreamsTopicStoreClusterTest {
+
+    @Test
+    public void testCluster(VertxTestContext context) throws Exception {
+        Assumptions.assumeTrue(KafkaStreamsTopicStoreTest.isKafkaAvailable());
+
+        KafkaStreamsConfiguration ksc1 = KafkaStreamsTopicStoreTest.configuration(Collections.emptyMap());
+        KafkaStreamsConfiguration ksc2 = KafkaStreamsTopicStoreTest.configuration(Collections.singletonMap(Config.APPLICATION_SERVER.key, "localhost:9001"));
+
+        TopicStore store1 = ksc1.getTopicStore();
+        TopicStore store2 = ksc2.getTopicStore();
+
+        Checkpoint async = context.checkpoint();
+
+        String topicName = "my_topic_" + ThreadLocalRandom.current().nextInt(Integer.MAX_VALUE);
+        Topic topic = new Topic.Builder(topicName, 2,
+                (short) 3, Collections.singletonMap("foo", "bar")).build();
+
+        Promise<Void> failedCreateCompleted = Promise.promise();
+
+        // Create the topic
+        store1.create(topic)
+                .onComplete(context.succeeding())
+
+                // Read the topic
+                .compose(v -> store2.read(new TopicName(topicName)))
+                .onComplete(context.succeeding(readTopic -> context.verify(() -> {
+                    // assert topics equal
+                    assertThat(readTopic.getTopicName(), is(topic.getTopicName()));
+                    assertThat(readTopic.getNumPartitions(), is(topic.getNumPartitions()));
+                    assertThat(readTopic.getNumReplicas(), is(topic.getNumReplicas()));
+                    assertThat(readTopic.getConfig(), is(topic.getConfig()));
+                })))
+
+                // try to create it again: assert an error
+                .compose(v -> store2.create(topic))
+                .onComplete(context.failing(e -> context.verify(() -> {
+                    assertThat(e, instanceOf(TopicStore.EntityExistsException.class));
+                    failedCreateCompleted.complete();
```

Comment:
Yeah, here we probably need to assert not only that it threw but that the state was not (or perhaps was) updated.

Reply:
How would this look in this vert.x code?